org.eclipse.lsp4e/src/org/eclipse/lsp4e/progress/LSPProgressManager.java

```diff
@@ -123,6 +125,7 @@ private void createJob(final LinkedBlockingDeque<ProgressParams> queue, final St
 			}
 		});
 		job.schedule();
+		jobs.add(job);

```

Comment:
The new job should have a listener to remove itself from the "jobs" when it's done.

Reply:
Thanks, I have done the change.